dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/OpenAPICorsTest.java

```diff
@@ -0,0 +1,124 @@
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.ws.microprofile.openapi.fat.utils.OpenAPIConnection;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import componenttest.topology.utils.HttpUtils.HTTPRequestMethod;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class OpenAPICorsTest {
+    private static final String SERVER_NAME = "CorsServer";
+    private static String REQUEST_HEADER_ORIGIN = "Origin";
+    private static String RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    private static String RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+    private static String RESPONSE_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers";
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests r = FATSuite.defaultRepeat(SERVER_NAME);
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        System.setProperty("sun.net.http.allowRestrictedHeaders", "true");
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    public void testEndpoint() throws Exception {
+        // Check that the /openapi endpoint is available.
+        final OpenAPIConnection conn = OpenAPIConnection.openAPIDocsConnection(server, false);
+        checkConnectionIsOK(conn);
+    }
+
+    @Test
+    public void testHeadersCorrectEndpoint() throws Exception {
+        // test CORS request to "http://localhost:8010/openapi" which is in CORS domain
+        OpenAPIConnection conn = OpenAPIConnection.openAPIDocsConnection(server, false);
+        conn.header(REQUEST_HEADER_ORIGIN, "openliberty.io");
+        HttpURLConnection response = conn.getConnection();
+
+        // check that CORS response headers are same as those defined in server.xml
+        assertEquals(RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_ORIGIN + " response header is incorrect or missing",
+            "openliberty.io", response.getHeaderField(RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertEquals(RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_CREDENTIALS + " response header is incorrect or missing",
+            "true", response.getHeaderField(RESPONSE_HEADER_ACCESS_CONTROL_ALLOW_CREDENTIALS));
+        assertEquals(RESPONSE_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS + " response header is incorrect or missing",
+            "TestHeader", response.getHeaderField(RESPONSE_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS));
+    }
```

Comment:
We should add a test that a pre-flight OPTIONS request is handled correctly and has the correct headers set.

Reply:
I've added an extra test for that